internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -155,13 +162,27 @@ func (p *clientgoCachedProxyResolver) ObjectExists(obj client.Object) (bool, err
 	return exists, err
 }
 
+func (p *clientgoCachedProxyResolver) IsReady() bool {
+	// If the proxy is has no database, it is only ready after a successful sync
+	// Otherwise, it has no configuration loaded
+	if p.dbmode == "off" {
+		p.configAppliedMutex.RLock()
+		return p.configApplied
+		p.configAppliedMutex.RUnlock()
```

Comment:
with above return statement, don't think p.configAppliedMutex.RUnlock() get executed.
should put p.configAppliedMutex.RUnlock() on top of return.
```suggestion
		p.configAppliedMutex.RUnlock()
		return p.configApplied
```